src/components/studies/app-design/AppDesign.tsx

```diff
@@ -676,16 +542,27 @@ const AppDesign: React.FunctionComponent<
 
   return (
     <Box className={classes.root}>
+      <NavigationPrompt when={hasObjectChanged} key="prompt">
```

Comment:
Added confirmation dialogue shown before the user attempts to navigate away from the page before saving.